backend/rest_test.go

```diff
@@ -0,0 +1,172 @@
+package backend_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/restic/restic/backend"
+	"github.com/restic/restic/backend/rest"
+)
+
+func setupRestBackend(t *testing.T) *rest.Rest {
+	url, _ := url.Parse("http://localhost:8000")
+	backend, _ := rest.Open(url)
+	return backend
+}
+
+func TestRestBackend(t *testing.T) {
+	// Initializing a temporary repository for the backend
+	path, _ := ioutil.TempDir("", "restic-repository-")
+	defer os.RemoveAll(path)
+
+	dirs := []string{
+		path,
+		filepath.Join(path, "data"),
+		filepath.Join(path, "snapshot"),
+		filepath.Join(path, "index"),
+		filepath.Join(path, "lock"),
+		filepath.Join(path, "key"),
+		filepath.Join(path, "temp"),
+	}
+
+	for _, d := range dirs {
+		os.MkdirAll(d, backend.Modes.Dir)
+	}
+
+	// Routing the repository requests
+	r := mux.NewRouter()
+
+	// Exists
+	r.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Most of this code can probably refactored into a working REST server, see my comment at https://github.com/restic/restic/pull/253#issuecomment-128130604